File: HaberUygulamaJava/app/src/main/java/com/example/haberuygulamajava/dao/HaberDao.java
package com.example.haberuygulamajava.dao;

import com.example.haberuygulamajava.model.HaberModel;
import com.example.haberuygulamajava.model.HaberTuruModel;
import com.example.haberuygulamajava.model.YorumInsertRequest;
import com.example.haberuygulamajava.model.YorumModel;
import com.example.haberuygulamajava.servis.ApiResponse;
import com.example.haberuygulamajava.servis.HaberService;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;

import java.util.List;

public class HaberDao {

    private HaberService service;

    public HaberDao(Retrofit retrofit) {
        service = retrofit.create(HaberService.class);
    }

    public void getHaberler(Callback<List<HaberModel>> callback) {
        service.getHaberler().enqueue(callback);
    }

    public void getYorumlar(int haberId, Callback<List<YorumModel>> callback) {
        service.getYorumlar(haberId).enqueue(callback);
    }

    public void insertYorum(YorumInsertRequest yorum, Callback<ApiResponse> callback) {
        service.insertYorum(yorum).enqueue(callback);
    }

    public void getKategoriler(Callback<List<HaberTuruModel>> callback) {
        service.getKategoriler().enqueue(callback);
    }

    public void getSon3Haber(Callback<List<HaberModel>> callback) {
        service.getSon3Haber().enqueue(callback);
    }

    public void getSonDakikaHaberler(Callback<List<HaberModel>> callback) {
        service.getSonDakikaHaberler().enqueue(callback);
    }

    public void getGundemHaberler(Callback<List<HaberModel>> callback) {
        service.getGundemHaberler().enqueue(callback);
    }
}

File: HaberUygulamaJetpack/app/src/main/java/com/example/haberuygulamajetpack/deo/HaberDao.kt
package com.example.haberuygulamajetpack.deo

import com.example.haberuygulamajetpack.model.HaberModel
import com.example.haberuygulamajetpack.model.HaberTuruModel
import com.example.haberuygulamajetpack.model.YorumInsertRequest
import com.example.haberuygulamajetpack.model.YorumModel
import com.example.haberuygulamajetpack.servis.ApiClient
import com.example.haberuygulamajetpack.servis.ApiResponse
import com.example.haberuygulamajetpack.servis.HaberService
import retrofit2.Retrofit
import retrofit2.awaitResponse


class HaberDao(var retrofit: Retrofit) {

    private val service = retrofit.create(HaberService::class.java)

    suspend fun getHaberler(): List<HaberModel>? {
        return try {
            val response = service.getHaberler().awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }


    suspend fun getYorumlar(haberId: Int): List<YorumModel>? {
        return try {
            val response = service.getYorumlar(haberId).awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }

    suspend fun insertYorum(yorum: YorumInsertRequest): ApiResponse? {
        return try {
            val response = service.insertYorum(yorum).awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }
    suspend fun getHaberById(haberId: Int): HaberModel? {
        return try {
            val response = service.getHaberById(haberId).awaitResponse()
            if (response.isSuccessful) response.body()!!.data else null
        } catch (e: Exception) {
            null
        }
    }


    suspend fun getKategoriler(): List<HaberTuruModel>? {
        return try {
            val response = service.getKategoriler().awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }
    suspend fun getSon3Haber(): List<HaberModel>? {
        return try {
            val response = service.getSon3Haber().awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }
    suspend fun getSonDakikaHaberler(): List<HaberModel>? {
        return try {
            val response = service.getSonDakikaHaberler().awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }
    suspend fun getGundemHaberler(): List<HaberModel>? {
        return try {
            val response = service.getGundemHaberler().awaitResponse()
            if (response.isSuccessful) response.body() else null
        } catch (e: Exception) {
            null
        }
    }


}

